### DISCUSSION TITLE
Command to start debug service hangs

### DISCUSSION THREAD

**Discussion by bhunsicker1** [codefori/vscode-ibmi#1143] [IBM i] [answered]
Hello

I am trying to see if I can automate the start of the debug service.   I understand it only needs to be started once and our delivery staff does not have the security to start it.  

If I use this command via either putty or QSH in a 5250 session, my session hangs.  It does appear to start the debug service.   

QSH CMD('/QIBM/Proddata/IBMiDebugService/bin/startDebugService.sh')

or via putty after changing to the directory, issue ./startDebugService.sh

I usually end up just killing my session. 

If I start this via vscode, vscode does not hang.   

I tried opening a case with IBM support, but they indicated they do not support this.  

Any suggestions?  

Thanks, 
Brenda

**Comment by chrjorgensen** [comment, marked as answer]
@bhunsicker1 The script in `/QIBM/Proddata/IBMiDebugService/bin/startDebugService.sh` actually starts the Java program, which is the debug service. So the debug service runs in your job, which then "hangs" (=does not end).

You should submit the QSHELL command to batch instead. Or add a ` &` to the end of your command in Putty, thus sending the command to a background task. Or you could use the excellent [Service Commander for IBM i](https://github.com/ThePrez/ServiceCommander-IBMi) tool to setup a service and control it using regular `STRTCPSVR` commands... many options... 😄

**Reply by bhunsicker1** [reply]
Thank you.
